core/src/test/java/com/klaviyo/coresdk/networking/NetworkBatcherTest.kt

```diff
@@ -0,0 +1,72 @@
+package com.klaviyo.coresdk.networking
+
+import android.content.Context
+import com.klaviyo.coresdk.KlaviyoConfig
+import com.nhaarman.mockitokotlin2.*
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class NetworkBatcherTest {
+    private val contextMock = mock<Context>()
+
+    @Before
+    fun setup() {
+        KlaviyoConfig.Builder()
+                .apiKey("Fake_Key")
+                .applicationContext(contextMock)
+                .networkFlushInterval(1000)
+                .networkFlushDepth(10)
+                .build()
+    }
+
+    @Test
+    fun `Network Batcher empties the queue when full`() {
+        val batcherSpy = spy<NetworkBatcher>()
+
+        doNothing().`when`(batcherSpy).initBatcher()
```

Comment:
can we use `.whenever` instead of `.'when'`?

Reply:
Good point, much clearner to keep using `whenever`